File: DailyChallenges/CPP/LeetCode_2405.cpp
class Solution {
public:
    int partitionString(string s) {
        set<char> ss;
        int ans = 0;
        for (char ch:s){
            if (ss.count(ch) != 0){ ss.clear(); ++ans; }
            ss.insert(ch);
        }
        return ans+1;
    }
};

File: DailyChallenges/Scala/LeetCode_2405.scala
object Solution {
    def partitionString(s: String): Int = {
        val chs = Array.fill[Int](26)(0)
        var ans = 1
        for (ch <- s){
            if (chs(ch-'a') == ans) ans += 1
            chs(ch-'a') = ans
        }
        ans
    }
}
